### PR TITLE
Project Metrics: Implement travis-flakiness metric

### PR DESCRIPTION
**Pull request by rcebulko**

### CODE REVIEW

File: project-metrics/metrics_service/metrics/travis_flakiness.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    if build_count == 0:
+      raise ValueError('No Travis builds to process.')
+    if build_count < 3:
+      raise ValueError('Not enough Travis builds to determine flakiness.')
+
+    flakes = 0
+    build_triples = zip(builds[:-2], builds[1:-1], builds[2:])
+    for prev_build, curr_build, next_build in build_triples:
+      if (curr_build.state != models.TravisState.PASSED and
+          prev_build.state == models.TravisState.PASSED and
```
Comment (danielrozenberg) [inline project-metrics/metrics_service/metrics/travis_flakiness.py]: Organize the conditions prev→curr→next, easier to read

File: project-metrics/metrics_service/metrics/travis_flakiness.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          next_build.state == models.TravisState.PASSED):
+        flakes += 1
```
Comment (danielrozenberg) [inline project-metrics/metrics_service/metrics/travis_flakiness.py]: If you reaaaaaaaaaaaaaaaally want to write "smart looking" code here (not required), make that if expression a separate function:

```python
_is_flaky(build_triples):
  prev_build, curr_build, next_build = build_triples
  return (prev_build.state == models.TravisState.PASSED and
      curr_build.state != models.TravisState.PASSED and
      next_build.state == models.TravisState.PASSED)
```

And then replace the for loop with:
```python
  flakes = sum(_is_flaky(build_triples) for build_triple in build_triples)
  # explicit conversion from True/False to 1/0 makes the sum correct
```

Comment (rcebulko) [reply project-metrics/metrics_service/metrics/travis_flakiness.py]: Yeah I thought about this but considered it to be over-engineering.

File: project-metrics/metrics_service/metrics/travis_flakiness.py (hunk 1 above)
Comment (rcebulko) [reply project-metrics/metrics_service/metrics/travis_flakiness.py]: Done

**Pull request merged**